piksi_multi_cpp/include/piksi_multi_cpp/observations/udp_observation_receiver.h

```diff
@@ -0,0 +1,46 @@
+
+#ifndef PIKSI_MULTI_CPP_OBSERVATIONS_UDP_OBSERVATIONS_RECEIVER_H_
+#define PIKSI_MULTI_CPP_OBSERVATIONS_UDP_OBSERVATIONS_RECEIVER_H_
+#include <piksi_multi_cpp/observations/observations_consumer.h>
+#include <atomic>
+#include <mutex>
+#include <thread>
+#include <vector>
+
+namespace piksi_multi_cpp {
+
+/*
+ * Class that receives corrections via UDP and can write them to a
+ * ObservationsConsumer.
+ *
+ * Together with ObservationsConsumer, This implements a visitor pattern,
+ * https://en.wikipedia.org/wiki/Visitor_pattern
```

Comment:
I guess we can follow this pattern to add any additional functionalities, e.g., base station sampling.